File: sdk-ts/src/authentication/apikey.ts
import { Credentials } from "./types";

export class ApiKeyAuth {
  private credentials: Credentials;
  private workspaceName: string;

  constructor(credentials: Credentials, workspaceName: string) {
    this.credentials = credentials;
    this.workspaceName = workspaceName;
  }

  async getHeaders(): Promise<Record<string, string>> {
    return {
      "X-Beamlit-Api-Key": this.credentials.apiKey || "",
      "X-Beamlit-Workspace": this.workspaceName,
    };
  }

  intercept(req: Request): void {
    req.headers.set("X-Beamlit-Api-Key", this.credentials.apiKey || "");
    req.headers.set("X-Beamlit-Workspace", this.workspaceName);
  }
}

File: sdk/apikey.go
package sdk

import (
	"context"
	"net/http"
)

type ApiKeyAuth struct {
	credentials   Credentials
	workspaceName string
}

func NewApiKeyProvider(credentials Credentials, workspaceName string) *ApiKeyAuth {
	return &ApiKeyAuth{credentials: credentials, workspaceName: workspaceName}
}

func (s *ApiKeyAuth) Intercept(ctx context.Context, req *http.Request) error {
	req.Header.Set("X-Blaxel-Api-Key", s.credentials.APIKey)
	req.Header.Set("X-Blaxel-Workspace", s.workspaceName)
	return nil
}
